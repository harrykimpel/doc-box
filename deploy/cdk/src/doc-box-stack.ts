import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { CreateDocumentFunc } from './doc/create-document-func';

export interface DocBoxStackProps extends StackProps {
}

export class DocBoxStack extends Stack {
  constructor(scope: Construct, id: string, props: DocBoxStackProps = {}) {
    super(scope, id, props);

    // The code that defines your stack goes here

    new CreateDocumentFunc(this, 'CreateDocumentFunc');
  }
}