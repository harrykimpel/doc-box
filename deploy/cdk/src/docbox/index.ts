export * from './document-bucket';
export * from './document-table';
export * from './document-api';
export * from './create-document-func';
export * from './attach-document-func';