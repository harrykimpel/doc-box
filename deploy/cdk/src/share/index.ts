export * from './go-base-func';