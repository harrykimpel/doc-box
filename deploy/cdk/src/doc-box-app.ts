import { App, Tags } from 'aws-cdk-lib';
import { DocBoxStack } from './doc-box-stack';

const app = new App();

Tags.of(app).add('domain', 'document-management');
Tags.of(app).add('owner', 'torenken');

new DocBoxStack(app, 'DocBoxStack');

app.synth();