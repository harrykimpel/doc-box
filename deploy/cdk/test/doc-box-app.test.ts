import { Template } from 'aws-cdk-lib/assertions';
import { DocBoxStack } from '../src/doc-box-stack';
import { TestApp } from './cdk-test-helper';


test('DocBoxStackSnapshot', () => {
  const app = new TestApp();
  const stack = new DocBoxStack(app, 'DocBoxStack');

  const template = Template.fromStack(stack);

  template.hasResourceProperties('AWS::Lambda::Function', {
    Environment: {
      Variables: {
        APPLICATION_CONTEXT: 'DocBox#CreateDocument',
      },
    },
  });

  expect(template.toJSON()).toMatchSnapshot();
});