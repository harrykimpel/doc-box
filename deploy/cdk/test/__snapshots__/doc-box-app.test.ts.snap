// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocBoxStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "CreateDocumentFunc54FAC593": Object {
      "DependsOn": Array [
        "CreateDocumentFuncServiceRole8188B1E2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "20b72f9687f0640363cd435b1d0511ad63039198d4731a1f7f592b1cf8c69aa0.zip",
        },
        "Environment": Object {
          "Variables": Object {
            "APPLICATION_CONTEXT": "DocBox#CreateDocument",
          },
        },
        "FunctionName": "doc-box-createDocument",
        "Handler": "bootstrap",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CreateDocumentFuncServiceRole8188B1E2",
            "Arn",
          ],
        },
        "Runtime": "provided.al2",
      },
      "Type": "AWS::Lambda::Function",
    },
    "CreateDocumentFuncLogRetention4967E25E": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "CreateDocumentFunc54FAC593",
              },
            ],
          ],
        },
        "RetentionInDays": 30,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "CreateDocumentFuncServiceRole8188B1E2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "c13434f8f1aa2ea30fa577b2feb208a41368b11787b752e10bfc71fe8eb919d5.zip",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
